### ISSUE TITLE
Test fails with increased block size

### ISSUE THREAD

**Issue by igorpisarev** [saalfeldlab/n5#12]
Just found that `N5Test` for reading/writing data blocks passes with the current test block size `[33,22,11]`, but fails for the data types that occupy >=4 bytes (int,long,float,double) if the block size is increased to `[44,33,22]`:
```
arrays first differed at element [16384]; expected:<1306016388> but was:<0>
	at org.junit.internal.ComparisonCriteria.arrayEquals(ComparisonCriteria.java:52)
	at org.junit.Assert.internalArrayEquals(Assert.java:416)
	at org.junit.Assert.assertArrayEquals(Assert.java:292)
	at org.junit.Assert.assertArrayEquals(Assert.java:305)
	at org.janelia.saalfeldlab.n5.N5Test.testWriteReadIntBlock(N5Test.java:221)
        ...
```
If the block size is increased to `[55,44,33]`, it fails for all supported data types with similar errors.

This happens in the most recent master and before the refactoring in the original implementation as well: https://github.com/saalfeldlab/n5/tree/55a57dadc9220f121f119099d90cc7bdb9d29c2b

**Comment by igorpisarev** [comment]
It somehow passes with the increased block size when the reading is implemented with `IOUtils.toByteArray(in)`, as I did in #6.
This makes me think that our code does not fully read the given buffer – will check that assumption.

**Comment by igorpisarev** [comment]
Confirmed that it was the case: the buffer needs to be populated in a loop with respect to how many bytes have actually been read from the input stream.
With this modification it passes all the tests. The only failing reader/writer was `lz4`, so it seems to depend on how the streams are implemented.
I will file a PR containing this fix soon.

**Comment by igorpisarev** [comment]
Fixed by merging #13.
